ambry-network/src/main/java/com/github/ambry/network/http2/Http2MultiplexedChannelPool.java

```diff
@@ -186,15 +188,17 @@ Http2ClientConfig getHttp2ClientConfig() {
       return promise.setFailure(new IOException("Channel pool is closed!"));
     }
 
-    if (parentConnections.size() >= http2ClientConfig.http2MinConnectionPerPort) {
-      // TODO: if warmup is not 100%, new connections are still required.
-      // This is a passive load balance depends on compareAndSet in claimStream().
-      for (MultiplexedChannelRecord multiplexedChannel : parentConnections) {
-        if (acquireStreamOnInitializedConnection(multiplexedChannel, promise)) {
-          return promise;
-        }
-        http2ClientMetrics.http2StreamSlipAcquireCount.inc();
+    List<MultiplexedChannelRecord> multiplexedChannelRecords = new ArrayList<>(parentConnections);
+
+    // Attempt at most multiplexedChannelRecords.size(). No slip acquire expected.
+    for (int i = 0; i < multiplexedChannelRecords.size(); i++) {
+      int randomIndex = random.nextInt(multiplexedChannelRecords.size());
+      MultiplexedChannelRecord multiplexedChannelRecord = multiplexedChannelRecords.get(randomIndex);
+      if (acquireStreamOnInitializedConnection(multiplexedChannelRecord, promise)) {
+        return promise;
       }
+      log.warn("Stream slip acquire on {}", inetSocketAddress);
+      http2ClientMetrics.http2StreamSlipAcquireCount.inc();
```

Comment:
nit: we can just Collections.shuffle(multiplexedChannelRecords, random, then go through the for loop.

Reply:
fixed.